internal/powerbiapi/client.go

```diff
@@ -33,20 +35,32 @@ func NewClientWithClientCredentialAuth(tenant string, clientID string, clientSec
 }
 
 func newClient(getAuthToken func(httpClient *http.Client) (string, error)) (*Client, error) {
-	httpClient := cleanhttp.DefaultClient()
+
```

Comment:
Most of this code was pulled over from another PR I had around upgrading the terraform version, the rest of that PR is ready to go in but added this part mostly so I could get a successful test run